app/code/Magento/ImportExport/Block/Adminhtml/Export/Filter.php

```diff
@@ -419,13 +419,23 @@ public function getRowUrl($row)
     }
 
     /**
-     * Prepare collection by setting page number, sorting etc..
-     *
      * @param \Magento\Framework\Data\Collection $collection
-     * @return \Magento\Eav\Model\ResourceModel\Entity\Attribute\Collection
+     * @return \Magento\Framework\Data\Collection
+     * @throws \Magento\Framework\Exception\LocalizedException
      */
     public function prepareCollection(\Magento\Framework\Data\Collection $collection)
     {
+        $_disabledAttrs = [
```

Comment:
My only concern is that the list of disabled attributes depends on the product type; that said, at the moment only simple products define that list. 
Furthermore, that list is not accessible, being defined in a protected variable.
We should have a way to collect all disabled attributes from all product types.
This implies a non-trivial code design change IMHO.
What do you think about this?
Thanks

Reply:
@aleron75 
Thanks for reviewing.

I think these are all disabled attributes for all product types. If you know others, please help.
To get  the list of disabled attributes , it should define as public var but it is protected .

Thank you.

Reply:
>I think these are all disabled attributes for all product types

I mean, in case of change we should maintain all of the lists; I would discourage making the list public because it would violate the encapsulation principle.

Probably a more recommended approach would consist in defining a provider class whose arguments are injectable via `di.xml` so that any other module can add its own:

```
<!-- module's di.xml -->
<config>
  ...
  <type name="Magento\CatalogImportExport\Model\Export\Product\DisabledAttributesProvider">
    <arguments>
      <argument name="disabledTypes" ssi:type="array">
        <item name="old_id" xsi:type="string">old_id</item>
        <item name="tier_price" xsi:type="string">tier_price</item>
        ...
    </arguments>
  </type>
  ...
</config>

At that point, the source of truth would become the provider class, whose `getList()` method would return the required list of disabled attributes.

Note: this would require a change in `\Magento\CatalogImportExport\Model\Export\Product\Type\AbstractType` and sublcasses because the functionality given by `getDisabledAttrs()` method and `protected $_disabledAttrs` property would be substituted by the new provider.

As I mentioned, it's not a trivial change: what do you think about this?

Reply:
I've asked for an opinion to the core team, let's see if we have some other feedback.

Reply:
Agree with @aleron75. We need to have a single place where we define a list of disabled attributes. For now this place is `\Magento\CatalogImportExport\Model\Export\Product\Type\AbstractType::getDisabledAttrs`. I suggest to verify if every product type needs the same list. If so, we can check all product types, then collect common attributes we need to hide.

Reply:
To sum up @irajneeshgupta, I think the easiest thing is taking inspiration from `\Magento\CatalogImportExport\Model\Export\Product::initTypeModels()` method: see there how for each product type, disabled products are collected.

You could inject `\Magento\ImportExport\Model\Export\ConfigInterface` via constructor to be able to get the needed `$productTypes` to collect all disabled attributes.

In this way you
1 - don't need to define another array
2 - collect all disabled attributes from different product types 

@kokoc does it sound good?

Reply:
@aleron75 
i have updated the way you suggested and it sounds really good :)